[[ Bug 18264 ]] Don't fail to build when unlicensed platform selected

livecodeali opened the pull request:
When unlicensed platforms are selected, an SB warning is added that
they were not licensed to build for those platforms.

If no licensed platforms are selected, the user is informed via an
error dialog and no further action is taken.

livecodeali committed:
[[ Bug 18264 ]] Don't fail to build when unlicensed platform selected
When unlicensed platforms are selected, an SB warning is added that
they were not licensed to build for those platforms.

If no licensed platforms are selected, the user is informed via an
error dialog and no further action is taken.

peter-b commented:
@livecode-vulcan review ok b1c715c

livecode-vulcan commented:
💙 review by @peter-b ok b1c715ceefab3332138e54285562106fe4b8c71a

peter-b commented:
@livecodeali Unfortunately it looks like this change is making the standalone-building tests fail on Travis.

livecode-vulcan commented:
😎 test success b1c715ceefab3332138e54285562106fe4b8c71a

* try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/659)
* try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/664)
* try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/659)
* try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/663)
* try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/668)
* try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/660)
* try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/660)

livecodeali committed:
[[ No UI ]] Don't remove unlicensed deploy platforms at pre-build stage
Because of the way the license parameters are set, we can't currently
check them in development command line environment. So just return true
at this stage and allow engine to throw on deploy.

peter-b commented:
@livecode-vulcan review ok cdb1a80

livecode-vulcan commented:
💙 review by @peter-b ok cdb1a80cc284dbbc6af7d2d2635f55b216047dae

livecode-vulcan commented:
😎 test success cdb1a80cc284dbbc6af7d2d2635f55b216047dae

* try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/662)
* try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/667)
* try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/662)
* try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/666)
* try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/671)
* try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/663)
* try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/663)
